app/src/main/java/com/example/hackernews/factories/RepositoryFactory.kt

```diff
@@ -0,0 +1,25 @@
+package com.example.hackernews.factories
+
+import com.example.hackernews.common.constants.Api
+import com.example.hackernews.data.service.NewsService
+import com.example.hackernews.model.repository.NewsRepository
+import okhttp3.OkHttpClient
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+object RepositoryFactory {
+    private val retrofit: Retrofit by lazy {
```

Comment:
Here we only create a single retrofit instance which will be used for creating different API clients (`NewsService` in our case)